pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -196,10 +202,38 @@ func changefeedPlanHook(
 
 			telemetry.Count(`changefeed.create.core`)
 			logChangefeedCreateTelemetry(ctx, jr)
-			err := distChangefeedFlow(ctx, p, 0 /* jobID */, details, progress, resultsCh)
-			if err != nil {
-				telemetry.Count(`changefeed.core.error`)
+
+			var err error
+			for r := retry.StartWithCtx(ctx, changefeedRetryOptions); r.Next(); {
```

Comment:
Let's move this logic to a separate function.. coreChangefeedWithRetry or some such.